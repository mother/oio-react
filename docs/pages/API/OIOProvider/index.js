import React from 'react'
import APITableBlock from '../../../components/APITableBlock'
import HeaderBlock from '../../../components/HeaderBlock'
import SpecimenBlock from '../../../components/SpecimenBlock'
import OIOProviderSpecimen from '../../../specimens/OIOProviderComponent'
import oioSpec from '../../../../src/OIOProvider/spec'

const OIOProviderPage = () => (
   <div>
      <HeaderBlock title="OIOProvider" />
      <SpecimenBlock specimen={OIOProviderSpecimen} />
      <APITableBlock spec={oioSpec} title="Component API" />
   </div>
)

export default OIOProviderPage
