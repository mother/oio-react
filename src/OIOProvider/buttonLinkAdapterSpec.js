export default [{
   name: 'isActive',
   type: 'Function',
   default: 'See Description',
   description: 'A function that is passed in a link. This function must return a boolean that represents whether the link is currently active. The default function does a simple check based on <code>window.location.pathname</code>. This is only by <code>ListMenuButton</code> and <code>TabMenuButton</code>, but not <code>Button</code>.'
}, {
   name: 'render',
   type: 'Function',
   default: 'See Description',
   description: 'A function that is passed the props passed in to the <code>Button</code> or <code>ListMenuButton</code> element. This function must return a React element. The default function simply returns <code>&lt;a href={linkTo}&gt;{children}&lt;/a&gt;</code>.'
}]
