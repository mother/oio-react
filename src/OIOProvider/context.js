import React from 'react'

const OIOContext = React.createContext()

const withOIOContext = WrappedComponent => React.forwardRef((props, ref) => (
   <OIOContext.Consumer>
      {context => (
         <WrappedComponent {...props} OIOContext={context} ref={ref} />
      )}
   </OIOContext.Consumer>
))

export { OIOContext as default, withOIOContext }
