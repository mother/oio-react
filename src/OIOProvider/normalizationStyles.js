import { css } from '@emotion/core'

const normalizationStyles = css`
   *{
      -webkit-tap-highlight-color: rgba(0,0,0,0);
      -webkit-tap-highlight-color: transparent; // "turn off" link highlight
   }

   *, *:before, *:after {
      -webkit-box-sizing: inherit;
      -moz-box-sizing: inherit;
      box-sizing: inherit;
   }

   html, body, div, span, applet, object, iframe,
   h1, h2, h3, h4, h5, h6, p, blockquote, pre,
   a, abbr, acronym, address, big, cite, code,
   del, dfn, em, img, ins, kbd, q, s, samp,
   small, strike, strong, sub, sup, tt, var,
   b, u, i, center,
   dl, dt, dd, ol, ul, li,
   fieldset, form, label, legend,
   table, caption, tbody, tfoot, thead, tr, th, td,
   article, aside, canvas, details, embed,
   figure, figcaption, footer, header, hgroup,
   menu, nav, output, ruby, section, summary,
   time, mark, audio, video {
      margin: 0;
      padding: 0;
      border: 0;
      vertical-align: baseline;
   }

   html {
     -webkit-box-sizing: border-box;
     -moz-box-sizing: border-box;
     box-sizing: border-box;
   }

   a {
      text-decoration: none;
      outline: none;
      border: none;
      cursor: pointer;
      -webkit-tap-highlight-color:rgba(0,0,0,0);
   }

   a:active {
      outline: none;
   }

   a:focus {
      -moz-outline-style: none;
      outline:0;
   }

   img {
      display: block;
      border: none;
      margin: 0;
      padding: 0;
   }

   table {
      margin: 0;
      padding: 0;
      border-collapse: collapse;
      border: none;
   }

   th, td {
      text-align: left;
      padding: 0;
   }

   td {
      vertical-align: top;
   }

   input, textarea {
      margin: 0;
      padding: 0;
      border: 0;
      outline: 0;
      border-radius: 0;
      -webkit-appearance: none;
   }
`

export default normalizationStyles
