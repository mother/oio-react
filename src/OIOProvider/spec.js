export default [{
   name: 'buttonLinkAdapter',
   type: 'Object',
   default: '-',
   description: 'See below'
}, {
   name: 'className',
   type: 'String',
   default: '-',
   description: 'Pass a <code>className</code> to customize className of the OIOProvider'
}, {
   name: 'fontFamily',
   type: 'String',
   default: '<code>sans-serif</code>',
   description: 'Default font-family used for all text inside this OIOProvider'
}, {
   name: 'fontSize',
   type: 'String',
   default: '<code>16px</code>',
   description: 'Default font-size used inside this OIOProvider component'
}, {
   name: 'id',
   type: 'String',
   default: '<code>oio-container-{timestamp}</code>',
   description: 'Container DOM Element ID'
}, {
   name: 'style',
   type: 'Object',
   default: '<code>{}</code>',
   description: 'Pass a <code>style</code> object to customize inline style of the OIOProvider'
}]
