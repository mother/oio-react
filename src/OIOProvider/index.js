// =======================================================
// OIOProvider
// All OIO components must be nested inside an OIOProvider
// You can nest OIOProvider inside other OIOProvider
// This component will be a Context provider in future PR
// =======================================================

import React from 'react'
/** @jsx jsx */
import { Global, jsx } from '@emotion/core'
import PropTypes from 'prop-types'
import OIOContext from './context'
import normalizationStyles from './normalizationStyles'

export default class OIOProvider extends React.Component {
   static propTypes = {
      children: PropTypes.node,
      className: PropTypes.string,
      fontFamily: PropTypes.string,
      fontSize: PropTypes.string,
      fontWeightLight: PropTypes.string,
      fontWeightNormal: PropTypes.string,
      fontWeightMedium: PropTypes.string,
      fontWeightSemibold: PropTypes.string,
      fontWeightBold: PropTypes.string,
      formElementAppearance: PropTypes.oneOf(['outline', 'plain']),
      formElementBorderColor: PropTypes.string,
      formElementBorderRadius: PropTypes.string,
      formElementBorderWidth: PropTypes.string,
      formElementBackgroundColor: PropTypes.string,
      formElementFocusBackgroundColor: PropTypes.string,
      formElementFocusBorderColor: PropTypes.string,
      formElementTextColor: PropTypes.string,
      formLabelTextColor: PropTypes.string,
      formLabelTextSize: PropTypes.string,
      formLabelTextTransform: PropTypes.string,
      formLabelTextWeight: PropTypes.string,
      highlightColor: PropTypes.string,
      style: PropTypes.object,
      textSizeScaleRatio: PropTypes.number,
      textSizeMultiplier: PropTypes.number
   }

   static defaultProps = {
      children: null,
      className: undefined,
      fontFamily: 'sans-serif',
      fontSize: '16px',
      fontWeightLight: '300',
      fontWeightNormal: '400',
      fontWeightMedium: '500',
      fontWeightSemibold: '600',
      fontWeightBold: '700',
      formElementAppearance: 'outline',
      formElementBorderColor: '#ddd',
      formElementBorderRadius: '3px',
      formElementBorderWidth: '2px',
      formElementBackgroundColor: 'transparent',
      formElementFocusBackgroundColor: 'rgba(0,0,0,0.01)',
      formElementFocusBorderColor: '#888',
      formElementTextColor: '#333',
      formLabelTextColor: '#444',
      formLabelTextSize: '0.8',
      formLabelTextTransform: 'uppercase',
      formLabelTextWeight: 'bold',
      highlightColor: '#2a383e',
      style: {},
      textSizeScaleRatio: 1.125,
      textSizeMultiplier: 1
   }

   render() {
      const {
         children,
         className,
         fontFamily,
         fontSize,
         fontWeightLight,
         fontWeightNormal,
         fontWeightMedium,
         fontWeightSemibold,
         fontWeightBold,
         formElementAppearance,
         formElementBorderColor,
         formElementBorderRadius,
         formElementBorderWidth,
         formElementBackgroundColor,
         formElementFocusBackgroundColor,
         formElementFocusBorderColor,
         formElementTextColor,
         formLabelTextColor,
         formLabelTextSize,
         formLabelTextTransform,
         formLabelTextWeight,
         highlightColor,
         style,
         textSizeMultiplier,
         textSizeScaleRatio
      } = this.props

      // Props available to OIO Context Consumer
      const contextProps = {
         fontFamily,
         fontSize,
         fontWeightLight,
         fontWeightNormal,
         fontWeightMedium,
         fontWeightSemibold,
         fontWeightBold,
         formElementAppearance,
         formElementBorderColor,
         formElementBorderRadius,
         formElementBorderWidth,
         formElementBackgroundColor,
         formElementFocusBackgroundColor,
         formElementFocusBorderColor,
         formElementTextColor,
         formLabelTextColor,
         formLabelTextSize,
         formLabelTextTransform,
         formLabelTextWeight,
         highlightColor,
         textSizeMultiplier,
         textSizeScaleRatio
      }

      return (
         <OIOContext.Provider value={contextProps}>
            <Global styles={normalizationStyles} />
            <div
               className={className}
               css={{ fontFamily, fontSize, ...style }}>
               {children}
            </div>
         </OIOContext.Provider>
      )
   }
}
