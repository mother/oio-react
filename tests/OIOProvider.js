import React from 'react'
import { render } from 'react-testing-library'
import OIOContext from '../src/OIOProvider/context'
import OIOProvider from '../src/OIOProvider'

test('OIO Provider provides context props correctly', () => {
   let contextProps = ''

   render(
      <OIOProvider>
         <OIOContext.Consumer>
            {(context) => {
               contextProps = context
               return (
                  <div>Hello World</div>
               )
            }}
         </OIOContext.Consumer>
      </OIOProvider>
   )

   expect(contextProps).toMatchSnapshot()
})

test('OIO Provider will render correctly', () => {
   const { baseElement } = render(
      <OIOProvider>
         Hello World
      </OIOProvider>
   )

   expect(baseElement).toMatchSnapshot()
})
