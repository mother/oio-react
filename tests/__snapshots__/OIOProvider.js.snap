// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OIO Provider provides context props correctly 1`] = `
Object {
  "fontFamily": "sans-serif",
  "fontSize": "16px",
  "fontWeights": Object {
    "bold": "700",
    "light": "300",
    "medium": "500",
    "normal": "400",
    "semibold": "600",
  },
  "textSizeMultiplier": 1,
  "textSizeScaleRatio": 1.125,
}
`;

exports[`OIO Provider will render correctly 1`] = `
.emotion-0 {
  font-family: sans-serif;
  font-size: 16px;
}

<body>
  <div>
    <div
      class=" emotion-0"
    >
      Hello World
    </div>
  </div>
</body>
`;
